languages/javalang/builder-renderer/commercetools-java-client/src/main/java/client/middlewares/LoggerMiddleware.java

```diff
@@ -0,0 +1,22 @@
+package client.middlewares;
+
+import client.ApiHttpHeaders;
+import client.Middleware;
+import client.MiddlewareArg;
+import json.CommercetoolsJsonUtils;
+
+import java.util.concurrent.CompletableFuture;
+
+public final class LoggerMiddleware implements Middleware {
+
+    final
+    @Override
+    public CompletableFuture<MiddlewareArg> next(MiddlewareArg arg) {
+        try {
+            System.out.println(CommercetoolsJsonUtils.toJsonString(arg));
```

Comment:
Here we should use a proper logging framework so that our users can decide how much gets logged. May be you can add a TDO here?

Reply:
sure, for me it was just for debugging purpose, but i would mark it as a todo 👍